### PR TITLE
Make ANCM Msis include version in name

### PR DESCRIPTION
**Pull request by jkotalik**
Fixes https://github.com/aspnet/AspNetCore/issues/13935.

I need to make a corresponding pr to VS once we insert ANCM again.

### CODE REVIEW

**Review by joeloff — COMMENTED** [review]
Just checking, but you don't need anything for CodeSigning to pick up the files? I'm guessing it's just doing *.msi or something

**Comment by jkotalik** [comment]
Yeah I believe so. I'll verify the output once it's done.

File: src/Installers/Windows/AspNetCoreModule-Setup/ANCMV2/AncmV2.wixproj [hunk 1]
```diff
@@ -7,7 +7,7 @@
     <IsShipping>true</IsShipping>
     <ProjectGuid>f9bacb48-3bd7-4ec2-ae31-664e8703ec12</ProjectGuid>
     <OutputType>Package</OutputType>
-    <OutputName>aspnetcoremodule_$(Platform)_en_v2</OutputName>
+    <OutputName>aspnetcoremodule_$(Platform)_en_v2_$(ANCMFolderVersion)</OutputName>
```
Comment (johnbeisner) [inline src/Installers/Windows/AspNetCoreModule-Setup/ANCMV2/AncmV2.wixproj]: Does 'ANCMFolderVersion' change on the same cadence as 'PackageVersion'?

Comment (jkotalik) [reply src/Installers/Windows/AspNetCoreModule-Setup/ANCMV2/AncmV2.wixproj]: It changes as frequently as the MSI version does (which is every day). That should be fine for our needs.

**Review by ghost — APPROVED** [review]

**Comment by jkotalik** [comment]
@joeloff this look good?

**Comment by johnbeisner** [comment]
@jkotalik 
Is the 'ANCMFolderVersion' discoverable? Meaning is it the Asp.NET version or discoverable from the Asp.NET version? Can you give us an example of the 'ANCMFolderVersion'

**Comment by jkotalik** [comment]
ANCMFolderVersion doesn't correspond with the ASP.NET version, it's similar but you can't the version from it. It has differences due to how ANCM was shipped before. Example is 13.0.19200. 

Do you need to know what the ANCMFolderVersion is? From what @joeloff told me, we don't. The ANCM msi isn't publically viewable, so this version on the msi file is purely for version upgrades working.

**Comment by johnbeisner** [comment]
When building the installer, we need to download the V2 module MSIs to the core-sdk build. We need to be able to discover the V2 module's version.

**Comment by jkotalik** [comment]
How does version discovery work today for other msis? Is there a props file I can add this version to?

**Comment by johnbeisner** [comment]
Can we use the build number in the asset name? 
Example: 
"aspnetcoremodule_$(Platform)_en_v2_3.1.0-preview1.19467.1.msi"

**Comment by johnbeisner** [comment]
Or something like this: 
`<PackageFileName>$(ANCMv2InstallerBaseName)-$(PackageVersion)-win-$(Platform)$(TargetExt)</PackageFileName>`

Where ANCMv2InstallerBaseName = "aspnetcoremodule_en_v2"

**Comment by jkotalik** [comment]
We can, it will just not match the actual ANCM msi version internally.

**Comment by anurse** [comment]
> Is the 'ANCMFolderVersion' discoverable?

We do have an `aspnetcoremodule.version` file in our build output with the full version number. Is this sufficient discoverability?

I'd like to keep these version numbers aligned with the real binary/msi/product versions. We have to version ANCM above 10.0 because of previous versions of ANCM that had versions like 8.x, etc.

**Comment by johnbeisner** [comment]
I've inspected the aspnetcoremodule.version file build-over-build and I can detect that the version doesn't change sometimes:

https://dotnetcli.blob.core.windows.net/dotnet/aspnetcore/Runtime/3.1.0-preview1.19467.1/aspnetcoremodule.version = 13.1.19260.0

https://dotnetcli.blob.core.windows.net/dotnet/aspnetcore/Runtime/3.1.0-preview1.19467.2/aspnetcoremodule.version = 13.1.19260.0

I believe this breaks the installer during upgrades for this component.

**Comment by jkotalik** [comment]
The version updates every day. Unless we ship MSIs multiple times in the same day, this won't break upgrade.

**Comment by jkotalik** [comment]
Resolved offline. We are going to use the dotnet version inside of the msi filename.

**Comment by anurse** [comment]
I talked to @jkotalik. We think the best solution in the end *is* to use the .NET Core version number in the file name. We need to keep the MSI version itself the same (`13.1.12345.0`) and we did *want* to keep the file name and MSI version aligned but I think given these requirements, it's probably best to just use the .NET Core version (`3.1.0-...`)

**Comment by johnbeisner** [comment]
Can you use the build/patch portion of the .NET Core version number in the ANCMFolderVersion. 
Example:
.NET Core version = 3.1.0-preview1.19467.2
ANCMFolderVersion = 13.1.19467.2 [both in the MSI and the asset name]

**Comment by jkotalik** [comment]
@johnbeisner trying. I believe you are right in that I can make that modification as the 19467 is larger than the highest version of ANCM (which is 19260).

@joeloff IIRC, the last number in an MSI version doesn't affect upgrade. Is that correct?

**Comment by joeloff** [comment]
@jkotalik I assume this takes care of the generated props file needed when inserting into VS as well?

**Comment by jkotalik** [comment]
> @jkotalik I assume this takes care of the generated props file needed when inserting into VS as well?

I don't believe it does yet. How can I verify the generated props file is there?

**Comment by johnbeisner** [comment]
@jkotalik 
@joeloff 
I think you need the last number for the ANCMFolderVersion - isn't that the build-of-the-day iterator? If you don't use the last integer; then the build at 8AM cannot be upgraded with your new build at 10am.

**Review by joeloff — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**